src/tools/rbd/action/Nbd.cc

```diff
@@ -173,9 +181,11 @@ int execute_detach(const po::variables_map &vm,
     return -EINVAL;
   }
 
-  std::vector<std::string> args;
+  if (device_name.empty() && vm.count(at::SNAPSHOT_ID)) {
```

Comment:
Nit: I would move this into existing `device_name.empty()` branch a few lines above.